swift/ql/lib/codeql/swift/elements/AstNode.qll

```diff
@@ -1,4 +1,6 @@
-// generated by codegen/codegen.py, remove this comment if you wish to edit this file
 private import codeql.swift.generated.AstNode
+private import codeql.swift.generated.Children
 
-class AstNode extends AstNodeBase { }
+class AstNode extends AstNodeBase {
+  AstNode getParent() { result = unique(AstNode x | this = getAChild(x) | x) }
```

Comment:
Like the `getAChild` predicate, I don't think this should be exposed on the AST classes.

In my mind there are two use-cases for `unique`:
- If something is functionally determined by its input, we can apply `unique` to tell the optimizer that there is in fact only one result. This helps the optimizer during join ordering (and probably a bunch of other things I don't know anything about)
- If something is in fact not functionally determined by its input and can have multiple results, but we only want it to have a result if there is one such result. For instance, if we'd like to find the unique `return` statement in a function or something like that.

Does the `unique` fit one of these cases? If so, I think we should add a comment here saying which case.

Reply:
I think it makes sense to move it to `Children.qll` 👍.

Reply:
> This comes from from a discussion on slack ([internal link](https://github.slack.com/archives/C02TU8JET9B/p1648643847232739?thread_ts=1648629027.395099&cid=C02TU8JET9B)), and as far as I can tell falls in the first case. I can add a comment.

Ah, right. It does indeed look like this is for the first case.